### PR TITLE
Post release 0.1.0: Update version to 0.1.0.dev0

### PR DESCRIPTION
**Pull request by srprash**
This PR prepares the main branch for the next development cycle by updating the version to 0.1.0.dev0 and updating the image version to be scanned to the latest released.
          
This PR should only be merge when release for version v0.1.0 is success.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/release_build.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
         with:
           image_uri_with_tag: "adot-autoinstrumentation-python:test"
           push_image: false
-          load_image: false
+          load_image: true
```
Comment (srprash) [inline .github/workflows/release_build.yml:35]: This change comes from the fix that we did on the `release/v0.1.x` branch: https://github.com/aws-observability/aws-otel-python-instrumentation/pull/160

Comment (XinRanZhAWS) [reply .github/workflows/release_build.yml:35]: was about to ask, thanks for call out

Comment (thpierce) [reply .github/workflows/release_build.yml:35]: Lets cut a backlog item that asserts that either `push_image` or `load_image` is true.

**Review by XinRanZhAWS — COMMENTED** [review]
There is a check fail, should we record and rerun?

**Review by thpierce — APPROVED** [review]

**Review by XinRanZhAWS — APPROVED** [review]

File: .github/workflows/release_build.yml (hunk 1 above)
Comment (srprash) [reply .github/workflows/release_build.yml:35]: done

**Pull request merged**